addons/stock/models/stock_lot.py

```diff
@@ -178,6 +179,14 @@ def _set_single_location(self):
         elif len(quants.location_id) > 1:
             raise UserError(_('You can only move a lot/serial to a new location if it exists in a single location.'))
 
+    @api.depends('quant_ids.location_id.usage', 'quant_ids.quantity')
+    def _compute_is_scrap(self):
```

Comment:
We should also ensure that the move linked to lot are scrap. Otherwise inventory adujstement will be consider as scrap